src/shared/index.ts

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ArtifactLocation, Log, Location, Region, Result } from 'sarif'
+
+type JsonLocation = { line: number, column: number } // Unused: pos
+type JsonRange = { value: JsonLocation, valueEnd: JsonLocation } // Unused: key, keyEnd
+export type JsonMap = Record<string, JsonRange>
+
+export type ResultId = [string, number, number]
+type _RegionBytes = [number, number] // byteOffset, byteLength
+type _RegionStartEndLineCol = [number, number, number, number] // start line, start col, end line, end col
+export type _Region
+	= number // single line
+	| _RegionBytes
+	| _RegionStartEndLineCol
+
+// Underscored members are optional in the source files, but required after preprocessing.
+declare module 'sarif' {
+	interface Log {
+		_uri: string
+		_uriUpgraded?: string
+		_jsonMap?: JsonMap
+		_augmented?: boolean
+		_distinct: Map<string, string> // Technically per Run, practically doesn't matter right now.
+	}
```

Comment:
So just a thought, I'm curious as to why you wouldn't create another object literal inside the Log interface (and the others as well) for the parsed results.

Something like.

interface Log {
  parsedResults?: {
      uri.: string,
     ...
  }
}

I think this would make the code easier to read when reading the code and trying to remember what "_" is really supposed to mean :)
That one comment will be super easy to miss.

Thoughts?

Reply:
I assume you mean 'parsed_______' here? Or do you really mean Sarif.Result (I don't see why)?

Given the most recent change to Typescript Strict, the optionality will *not* be "time-dependent" (aka before or after processing/augmentation). Now fields can be truly non-optional (or truly optional). And in this evolved world, I would argue we could drop the underscores (aside from a few naming collisions). In practice I don't think there's anywhere we behave differently (or need to know) if the property is processed/augmented/parsed/etc or not. The distinction might not be required.

---
In reply to: [437557263](https://github.com/microsoft/sarif-vscode-extension/pull/279#discussion_r437557263) [](ancestors = 437557263)

Reply:
"By Design" for now. Feel free to re-open if you disagree.

---
In reply to: [437659428](https://github.com/microsoft/sarif-vscode-extension/pull/279#discussion_r437659428) [](ancestors = 437659428,437557263)